registry/interface.go

```diff
@@ -18,6 +21,6 @@ type SignatureRepository interface {
 	// Put uploads the signature to the registry
 	Put(ctx context.Context, signature []byte) (notation.Descriptor, error)
 
-	// Link creates an signature artifact linking the manifest and the signature
-	Link(ctx context.Context, manifest, signature notation.Descriptor) (notation.Descriptor, error)
+	// CreateSignatureManifest creates an signature artifact linking the manifest and the signature
+	CreateSignatureManifest(ctx context.Context, manifest, signature notation.Descriptor) (notation.Descriptor, error)
```

Comment:
How about combining `Put` and `Link` into one?
```go
Put(ctx context.Context, signature []byte, annotations map[string]string) (notation.Descriptor, SignatureManifest, error)
```

where the returned descriptor is the descriptor of the signature manifest?

Reply:
Since `CreateSignatureManifest` is also doing something similar to `put`, it's reasonable to me to have them combined.

Reply:
Done